tools/retrive_go_repos.py

```diff
@@ -0,0 +1,302 @@
+        if "github.com/" in github_url and \
+                github_url not in dependancy_urls and \
+                is_valid_repo(get_repo_details(github_url.split("github.com/")[1])):
+            dependancy_urls.append(github_url)
+
+
+def get_go_pkg_github_repo_details(pkg: str):
+    """Get Go package and repo details."""
+    if pkg not in pkg_list:
+        pkg_list.append(pkg)
+
+        data = get_go_pkg_data(pkg)
+        if 'Repository: <a href="https://github.com/' in data.text:
+            get_dependancy_repo_url_from_text(data.text)
+
+
+def is_valid_repo(item) -> bool:
+    """Validate repo is valid to consider or not."""
+    if item:
+        try:
+            if item and item['archived'] is False and \
+                    check_repo_updated_date(item['updated_at']) and \
+                    str(item['language']).upper() == LANGUAGE and \
+                    item['disabled'] is False:
+                return True
+        except RuntimeError:
+            logging.error("Error while validating repo details : {repo}".format(repo=item['full_name']))
+    return False
+
+
+def get_repo_details(org_repo: str):
+    """Get github repo details."""
+    url = "https://api.github.com/repos/{org_repo}".format(org_repo=org_repo)
+    try:
+        result = requests.get(url, headers={'Authorization': 'Bearer {token}'.format(token=GITHUB_ACCESS_TOKEN)})
+        if result.status_code == 200:
+            return result.json()
+        else:
+            logging.error("Repo doesn't exist : {repo}".format(repo=org_repo))
+    except Exception as ex:
+        logging.error("Error while retriving repo details : {repo}, msg: {msg}".format(repo=org_repo, msg=str(ex)))
+
+    return None
+
+
+def remove_unwanted_chars(line) -> str:
+    """Remove unwanted characters from string."""
+    return str(line).replace("\\t", "").replace("\\n'", "").replace("b'", "").strip()
+
+
+def get_dependancy_data_from_go_mod_file(org_repo: str):
+    """Get Dependancy repo details from go.mod file."""
+    content_raw_url = "https://raw.githubusercontent.com/{org_repo}/master/go.mod".format(org_repo=org_repo)
+    dependancy_section_started = False
+    dependancy_section_ended = False
+
+    for line in urlopen(content_raw_url):
+        str_line = remove_unwanted_chars(line)
+        if dependancy_section_started is False and "require" in str_line:
+            dependancy_section_started = True
+
+        if ")" in str_line:
+            dependancy_section_ended = True
+
+        if dependancy_section_ended and dependancy_section_started:
+            break
+
+        splited_text = str_line.split(" ")
+        if len(splited_text) > 1:
+            pkg = splited_text[0].strip()
+            get_go_pkg_github_repo_details(pkg)
+
+
+def get_dependancy_data_from_lock_file(org_repo: str):
+    """Get Dependancy repo details from Gopkg.lock file."""
+    content_raw_url = "https://raw.githubusercontent.com/{org_repo}/master/Gopkg.lock".format(org_repo=org_repo)
```

Comment:
Can we move all URLs to a single dict in one function? It would be much easier to manage once the same URL is required in multiple locations.

Reply:
done